src/FakeItEasy/Configuration/BuildableCallRule.cs

```diff
@@ -11,7 +11,7 @@ namespace FakeItEasy.Configuration
     internal abstract class BuildableCallRule
         : IFakeObjectCallRule
     {
-        private readonly List<WherePredicate> wherePredicates;
+        private readonly SynchronizedCollection<WherePredicate> wherePredicates;

```

Comment:
Well, with that reasoning, all frameworks would only use thread-safe collections everywhere... But thread-safety isn't free, so it should only be applied where it makes sense.